tools/debian/convert_debian.py

```diff
@@ -34,97 +36,179 @@
+                current_advisory, dsa_match.group(2))
+
+
+def write_output(output_dir: str, advisories: dict[str, AdvisoryInfo]):
+    for key in advisories:
+        output_file = open(os.path.join(output_dir, key + ".json"), "w")
+        output_file.write(str(advisories[key]))
+        print("Writing: " + os.path.join(output_dir, key + ".json"), flush=True)
+
+    print("Complete")
+
+
+def convert_debian(webwml_repo: str, security_tracker_repo: str, output_dir: str):
+    """Convert Debian advisory data into OSV."""
+    advisories: dict[str, AdvisoryInfo] = {}
+
     # Enumerate advisories + version info from security-tracker.
     with open(os.path.join(security_tracker_repo,
                            SECURITY_TRACKER_DSA_PATH)) as handle:
-        for line in handle:
-            line = line.rstrip()
-            if not line:
-                continue
-
-            if LEADING_WHITESPACE.match(line):
-                # Within current advisory.
-                if not current_advisory:
-                    raise ValueError('Unexpected tab.')
-
-                # {CVE-XXXX-XXXX CVE-XXXX-XXXX}
-                line = line.lstrip()
-                if line.startswith('{'):
-                    advisories[current_advisory].cves = line.strip('{}').split()
-                    continue
-
-                if line.startswith('NOTE:'):
-                    continue
-
-                version_match = VERSION_PATTERN.match(line)
-                if not version_match:
-                    raise ValueError('Invalid version line: ' + line)
-
-                advisories[current_advisory].versions.append(
-                    VersionInfo(version_match.group(1), version_match.group(2),
-                                version_match.group(3)))
-            else:
-                if line.strip().startswith('NOTE:'):
-                    continue
-
-                # New advisory.
-                dsa_match = DSA_PATTERN.match(line)
-                if not dsa_match:
-                    raise ValueError('Invalid line: ' + line)
-
-                current_advisory = dsa_match.group(1)
-                advisories[current_advisory] = AdvisoryInfo(
-                    current_advisory, dsa_match.group(2))
-
-    print(advisories)
+        parse_security_tracker_file(advisories, handle)
+
+    file_path_map = {}
+
+    for root, dirs, files in os.walk(os.path.join(webwml_repo, WEBWML_SECURITY_PATH)):
+        for file in files:
+            file_path_map[file] = os.path.join(root, file)
+
+    # Add descriptions to advisories from wml files
+    for key in advisories:
+        # TODO: use regex to remove suffix?
+        mapped_key_no_ext = key.lower().removesuffix('-1').removesuffix('-2').removesuffix('-3').removesuffix('-4')
+        val_wml = file_path_map.get(mapped_key_no_ext + ".wml")
+        val_data = file_path_map.get(mapped_key_no_ext + ".data")
+
+        if not val_wml:
+            print(mapped_key_no_ext)
```

Comment:
Is this debugging code? can remove if so.

Reply:
The .wml and .data files seem to take a bit of time to generate and get placed in the repository. So for the latest few DSA's the corresponding .wml file will not exist. I'm not sure what the desired behavior should be for this, either:

- Just generate a partial .json file
- Not generate anything for now and skip the latest DSA's

For now I updated it to emit a descriptive message about the wml file not existing.

Reply:
Let's generate a partial .json file, and fill the description/summary with boilerplate (e.g. "Details not yet available").

Reply:
Added a TODO there for now